src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -2610,8 +2620,8 @@ public BigInteger pow(int exponent) {
 
         // Factor the powers of two out quickly by shifting right, if needed.
         if (powersOfTwo > 0) {
```

Comment:
`shiftRight()` returns `this` when the shift distance is 0, so there's no compelling need to have two branches for `powersOfTwo`